modules/decline/src/core/OptsVisitor.scala

```diff
@@ -326,5 +340,9 @@ object OptsVisitor
     schema.compile[Opts](this).map(bijection.to)
 
   override def refine[A, B](schema: Schema[A], refinement: Refinement[A, B]) =
-    schema.compile[Opts](this).map(refinement(_).fold(message =>throw RefinementFailed(message),identity))
+    schema
+      .compile[Opts](this)
+      .map(
+        refinement(_).fold(message => throw RefinementFailed(message), identity)
+      )
 }
```

Comment:
**thought:** can we use `mapValidated` or similar to, instead of throwing, fail the Opts value?